packages/gatsby-theme-docs/src/layouts/internals/layout-application.js

```diff
@@ -41,18 +43,55 @@ const Container = styled.div`
   }
 `;
 
-const LayoutApplication = (props) => (
-  <>
-    <Root
-      role="application"
-      id="application"
-      isGlobalNotificationVisible={Boolean(props.globalNotification)}
-    >
-      <Container {...props} />
-    </Root>
-    <div id="modal-portal" />
-  </>
-);
+const localStorageProvider = () => {
+  // When initializing, we restore the data from `localStorage` into a map.
+  const map = new Map(JSON.parse(localStorage.getItem('app-cache') || '[]'));
+
+  const handleBeforUnload = (event) => {
+    const appCache = JSON.stringify(Array.from(map.entries()));
```

Comment:
This handler is very aggressive as it will dump swr state to the cache each time the page is refreshed. But we refresh the page alsow each time we login and logout.
This is the reason of the 2 custom events that basically disable writing the cache upon login and logout.